src/utils/options.ts

```
@@ -240,6 +245,7 @@ export {
   agentPort,
   connConnectTime,
   recoveryCodeFile,
+  recoveryCodeOutFile,
```

What would be the problem with re-usung the original option `recoveryCodeFile`? Is there fundamental ambiguity here?

---

We don't want to reuse the input recovery code file path, as that means it may try to overwrite the existing recovery code file. Additionally, the user may want to customise the output directory so it makes sense to keep these separate.

---

Its better that the difference is explicit. While yes, we can't have an input and an output function at the same time, How would we tell the input apart from output? Check that we're imputing a path rather than a BIP39 code? Switching that way is way to implicit for my liking.

---

If one were to set `--recovery-code-file=/path/to/somewhere`.

Then it could mean that the program is going to read from that path.

It could meant that the program will write to the path.

In what situation where it would ever be the case where you'd want to read from path A, and then write to path B?

As far as I know atm, in every situation where you'd want to read a recovery code, or write to the recovery code, you'd want to have the same path. In fact it makes sense as the option that designates the location of the recovery code.

However disregard what I said **IF** the presence of the option `--recovery-code-file` actually changes the behaviour of polykey agent start. If this is true, such as the case where it tries to start by reading the recovery code... then maybe it should be 2 options.

If there are 2 options - it has to be documented clearly which option is actually relevant. I can imagine ambiguity here:

```
# is this about starting an agent, recovery with A, and then writing it out to B?
polykey agent start --recovery-code-file=/path/to/A --recovery-code-out-file=/path/to/B

# is this about writing out to B a generated recovery code? (What if the key already existed?)
polykey agent start --recovery-code-out-file=/path/to/B

# what about this?
polykey agent start --recovery-code-file=/path/to/B
```

I think we have to make a decision here. When an option is not used, should we throw an error to indicate to the user that the option does not make sense here? Or should we just ignore the option - and we could even warn on stderr that the option is not being used - due to _reasons_. I'm leaning to the latter atm.

---

I'll make sure that they are nicely documented. The `recovery-code-out-file` option by necessity only applies when needed. When not needed it is just ignored. For it's usage there are two cases

1. node generated a new recovery code for whatever reason - so rather than printing it out we divert the recovery code to a file specified by that path.
2. Node started with existing state - Do nothing, there was no recovery code to output. 

For case 2, I'd rather just have it silently do nothing. The service will have a static configuration. So first time it starts it needs to not output the code to the logs, but for all subsequent starts we don't need to handle it. But in both the initial and subsequent starts the `recovery-code-out-file` will be set.


This option needs to be separate from the `recovery-code-file` since their intentions *will* conflict depending on state. Starting a node without state will need a clear difference between inputting a recovery code, or outputting it to a file.

---

I've updated the description to indicate that one is for reading the recovery code input, and the other is only used to divert the recovery code output.